tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -478,3 +481,137 @@ def _make_tf_features(self, input_feat):
                                   [-1, self._feature_size])
       freq_inputs.append(cur_input)
     return freq_inputs
+
+class MultiGFRNNCell(rnn_cell.MultiRNNCell):
+  """Gated Feedback RNN MultiCell implementation based on:
+
+  Junyoung Chung, Caglar Gulcehre, Kyunghyun Cho, Yoshua Bengio
+  Gated Feedback Recurrent Neural Networks, 2015
+  https://arxiv.org/abs/1502.02367
+
+  It differs from MultiRNNCell because it passes the whole state of the previous
+  hidden layer inside each cell. It only accepts GFLSTMCells or wrapped GFLSTMCells.
+  """
+
+  def __init__(self, cells, state_is_tuple=False):
+    """Create a RNN cell composed sequentially of a number of GFLSTMRNNCells.
+
+    Args:
+      cells: list of GFLSTMCells that will be composed in this order.
+      state_is_tuple: If True, accepted and returned states are n-tuples, where
+        `n = len(cells)`.  By default (False), the states are all
+        concatenated along the column axis.
+
+    Raises:
+      ValueError: if cells is empty (not allowed), or at least one of the cells
+        returns a state tuple but the flag `state_is_tuple` is `False` or one
+        of the cells is not a GFLSTMCell
+    """
+
+    # TODO Must check somehow that the cells are instances of GFLSTMCell
+    # even if they're wrapped
+    # Can potentially accept more types of GF cells if they get implemented.
+    # GFGRUCell or GFRNNCell are candidates.
+    for cell in cells:
+      if not isinstance(cell, GFLSTMCell):
+        raise ValueError("Cells must be of type GFLSTMCell")
+
+      super().__init__(cells, state_is_tuple)
+
+  def __call__(self, inputs, state, scope=None):
+    """Run the multi-layer cell on inputs, starting from state.
+    The hidden hs from state get passed inside each cell.
+    """
+    with vs.variable_scope(scope or type(self).__name__):  # "MultiGFLSTMCell"
+      curr_state_pos = 0
+      prev_hs = []
+      prev_states = []
+      # We accumulate the prev_hs once here so we don't split them again in each layer
+      # We also accumulate the prev_states since we're already splitting
+      for i, cell in enumerate(self._cells):
+        with vs.name_scope("Cell%d" % i):
+          if self._state_is_tuple:
+            if not _is_sequence(state):
+              raise ValueError(
+                  "Expected state to be a tuple of length %d, but received: %s"
+                  % (len(self.state_size), state))
+            prev_state = state[i]
+          else:
+            prev_state = array_ops.slice(state, [0, curr_state_pos], [-1, cell.state_size])
+            curr_state_pos += cell.state_size
+          if cell._state_is_tuple:
```

Comment:
This part will only work properly for some cell types, namely LSTM and I think GRU as well. To make it more general we could unpack inside the cell itself, but that means more unpacks.

It also breaks when wrappers are used